### PR TITLE
Enable js e2e tests

### PR DESCRIPTION
**Pull request by ericlewis**
## Summary

Part of #23561. Small refactor to use 1 marker for the different tests, as it doesn't matter where it is- so long as we can detect it.

## Changelog

[General] [Fixed] - turn on JS e2e tests

## Test Plan

run `node ./scripts/run-ci-e2e-tests.js --js`

### CODE REVIEW

**Comment by ericlewis** [comment]
I removed the flow checks here, as they occur earlier on too. Is it important for us to flow check here too? (Or jest?)

**Comment by hramos** [comment]
Hmm, given this script is meant to run on CI, and we already run Flow check elsewhere, I think it'd be fine to drop it from this script.

**Comment by hramos** [comment]
(Also wanted to note, I'm immensely grateful you're tackling this!)

**Comment by hramos** [comment]
That Switch test has been super flaky in the past. Feel free to remove the entire Switch-test.js file if you need to. See https://github.com/facebook/react-native/pull/23410 where I disabled one of the Switch tests just this past month.

**Comment by ericlewis** [comment]
@hramos cool, I will take a look when I get back to keyboard.

File: .circleci/config.yml [hunk 1]
```diff
@@ -245,6 +245,10 @@ aliases:
     name: iOS End-to-End Test Suite
     command: node ./scripts/run-ci-e2e-tests.js --ios --retries 3;
 
+  - &run-js-e2e-tests
+    name: JavaScript End-to-End Test Suite
+    command: node ./scripts/run-ci-e2e-tests.js --js --retries 3;
+
   - &display-broken-tests-warning
```
Comment (hramos) [inline .circleci/config.yml]: This is unrelated to your PR but if you can tackle it while you're touching this: can you remove this alias? It's no longer used.

**Pull request closed without merging**